onnxruntime/python/tools/transformers/fusion_rotary_attention.py

```diff
@@ -592,6 +739,14 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
             self.nodes_to_remove.append(k_nodes[1])
             self.nodes_to_remove.append(k_nodes[3])
             self.nodes_to_remove.append(k_nodes[4])
+        elif k_nodes == k_nodes_4:
+            remove_dic = {}
+            node_keep_names = [k_nodes[0][-1].name, k_nodes[0][-4].name]
+            for temp_path in k_nodes:
+                for temp_node in temp_path:
+                    if temp_node.name not in remove_dic and temp_node.name not in node_keep_names:
```

Comment:
Same question for removing some of the condition checks and using `add_nodes_to_remove` instead